pkg/ccl/backupccl/restore_old_versions_test.go

```diff
@@ -730,3 +744,44 @@ func TestRestoreWithDroppedSchemaCorruption(t *testing.T) {
 	}
 	require.Falsef(t, hasSameNameSchema(dbName), "corrupted descriptor exists")
 }
+
+// restorePublicSchemaRemap tests that if we're restoring a database from
+// an older version where the database has a synthetic public schema, a real
+// descriptor backed public schema is created and the tables in the schema
+// are correctly mapped to the new public schema.
+func restorePublicSchemaRemap(exportDir string) func(t *testing.T) {
```

Comment:
I would prefer to also see tests for a cluster restore, and a table restore into an existing database since that stresses different descriptor rewrite code paths.

Reply:
I added a check in `restoreOldVersionClusterTest` to verify that the public schema has correct ids.

```
		// In the backup, Public schemas for non-system databases have ID 29.
		// These should all be updated to explicit public schemas.
		sqlDB.CheckQueryResults(t, "SELECT * FROM system.namespace WHERE name='public'", [][]string{
			{"1", "0", "public", "29"},
			{"50", "0", "public", "64"},
			{"51", "0", "public", "65"},
			{"52", "0", "public", "66"},
		})
```

Reply:
Also added a test for a table restore in this test.
```
		// Test restoring a single table and ensuring that d.public.t which
		// previously had a synthetic public schema gets correctly restored into the
		// descriptor backed public schema of database test.
		sqlDB.Exec(t, `CREATE DATABASE test`)
		sqlDB.Exec(t, `RESTORE d.public.t FROM $1 WITH into_db = 'test'`, LocalFoo)

		sqlDB.CheckQueryResults(t, `SELECT x FROM test.public.t`, [][]string{{"3"}, {"4"}})
```

Reply:
Thank you!